src/programs/solo12_hardware_calibration.cpp

```diff
@@ -89,13 +90,17 @@ int main(int argc, char** argv)
     shared_content.sc_mutex.unlock();
 
     // print the home offset:
-    while (!robot_ready)
+    do
```

Comment:
here you could have while loop before to "wait until ready", I guess the solo have that.

Reply:
I can give it a try.  In this case we would be missing the lock around the "is ready" check, though.  But I'm not sure if this is actually needed here.